articlequality/extractors/extractor.py

```diff
@@ -74,20 +80,16 @@ def extract(self, page, verbose=False):
 
                 if revert is not None:
                     # This revision is a revert.
-                    for rev_id in revert.reverteds:
-                        if rev_id in labelings:
-                            for lab in labelings[rev_id]:
-                                lab['reverted'] = True
-
-                    if revert.reverted_to in labelings:
-                        for lab in labelings[revert.reverted_to]:
-                            lab['reverted'] = False
-
+                    self.invert_reverted_status(
+                        revisions[revision.id]['reverted'],
+                        revisions, labelings)
                     if verbose:
                         sys.stderr.write("r")
                         sys.stderr.flush()
                 else:
                     # This revision is not a revert.  Get the new labels
+                    # FIXME: what if last_labels contains labels from one of
+                    # reverted edits?
                     new_labels = project_labels - last_labels
```

Comment:
Since we're storing every single revision in `revisions`, we could possibly also store the labels and take a second pass looking for revisions that are not reverted.